static/app/views/performance/newTraceDetails/index.tsx

```diff
@@ -338,6 +349,26 @@ export function TraceViewWaterfall(props: TraceViewWaterfallProps) {
     throw new Error('Invalid trace state');
   }, [props.traceSlug, props.trace, props.status, projects, props.replayRecord]);
 
+  useEffect(() => {
+    if (props.replayTraces && props.replayTraces.length > 0 && tree.type === 'trace') {
+      tree.fetchTraces({
+        api,
+        filters,
+        replayTraces: props.replayTraces,
+        organization: props.organization,
+        traceLimit: undefined,
+        urlParams,
+        rerender: forceRerender,
+      });
+    }
+
+    return () => {
+      tree.stopFetching();
+    };
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
We should remove this and add all deps. The if condition inside the effect should be the place where we determine if we want to fetch this or not